### PR TITLE
Display selected datasets on the map

### PR DESCRIPTION
**Pull request by danielfdsilva**
Done so far:
- Displays selected datasets on the map
- Dataset order support
- Dataset opacity support

Compare is not possible yet.

### CODE REVIEW

File: app/scripts/components/common/map/utils.ts [hunk 1]
```diff
@@ -123,3 +122,63 @@ export async function requestQuickCache({
...
+type Res<T> = T extends Fn
+  ? T extends DatasetDatumFn<DatasetDatumReturnType>
+    ? DatasetDatumReturnType
+    : never
+  : T extends any[]
+  ? Res<T[number]>[]
+  : T extends object
+  ? ObjResMap<T>
+  : T;
+
```
Comment (nerik) [inline app/scripts/components/common/map/utils.ts:140]: Move to `types.d.ts` ?

File: app/scripts/components/common/map/utils.ts [hunk 2]
```diff
@@ -123,3 +122,63 @@ export async function requestQuickCache({
...
+export function resolveConfigFunctions<T>(
+  datum: T,
+  bag: DatasetDatumFnResolverBag
+): Res<T>;
+/* eslint-disable-next-line no-redeclare */
```
Comment (nerik) [inline app/scripts/components/common/map/utils.ts:145]: No need for this AFAIK

File: app/scripts/components/common/map/utils.ts [hunk 3]
```diff
@@ -123,3 +122,63 @@ export async function requestQuickCache({
...
+export function resolveConfigFunctions<T extends any[]>(
+  datum: T,
+  bag: DatasetDatumFnResolverBag
+): Res<T[number]>[];
+/* eslint-disable-next-line no-redeclare */
+export function resolveConfigFunctions(
```
Comment (nerik) [inline app/scripts/components/common/map/utils.ts:151]: TBH I'm struggling to understand what this does. `resolveConfigFunctions` can call itself if datum is iterable, that I get. but then in other situations, it calls datum as a function? How is that related to the call in index.tsx line 220? What does `bag` mean exactly? 

Sorry, might just be a case of Friday afternoon.

**Review by nerik — CHANGES_REQUESTED** [review]
It works 🙌 

I guess my main comments are around `resolveConfigFunctions`, and also layer order:
1. the stacking order on the map is the opposite of the one in the timeline. As in, 'up' in the list should translate to 'on top' in the map, at least that's how I've always seen that kind of stuff work.
2. this is due to the sorting algorithm in styles.tsx, hence 100% my legacy, but I think we should either:
  - follow the natural order of generators in their array, and not explicitly declare order in the raster-timeseries signature (in that case the sorting logic should use the generators indices, not the id)
  - keep explicitly declaring order, but in that case use that numerical order directly in styles.tsx, rather than the string id. Also, in that case, no need to declare and use the order prop in raster-timeseries, but put it in the `BaseGeneratorParams` type and let styles.tsx handle this
  
This can be merged without addressing 2. which I can do myself later.

File: app/scripts/components/common/map/utils.ts (hunk 3 above)
Comment (danielfdsilva) [reply app/scripts/components/common/map/utils.ts:151]: This basically resolves the functions in the dataset config and is needed to support this behavior.
The [docs](https://github.com/NASA-IMPACT/veda-ui/blob/main/docs/content/frontmatter/layer.md#function-values) show some examples and explain the behavior. If it is not clear, we should improve the docs.

This was used in the past for the customization of the compare map message. Although it is a neat little power user functionality, not sure it is used (known/needed) a lot.

**Comment by danielfdsilva** [comment]
@nerik I think that 2.1 (layer's own index order) is what makes most sense.

File: app/scripts/components/common/map/utils.ts (hunk 1 above)
Comment (danielfdsilva) [reply app/scripts/components/common/map/utils.ts:140]: These types are only ever used for the `resolveConfigFunctions` function declaration and are not needed elsewhere so I think they can live with the declaration. However feel free to move them if you like

**Comment by danielfdsilva** [comment]
@nerik Did some cleanup to and moved the map code to a separate component.
The compare now can be enabled through the mock controls and shows the correct datasets with a date set 1 month from the main date.
If you want to address the layer order issue before merging, feel free to continue to work on this branch.

**Comment by nerik** [comment]
@danielfdsilva Merged this for now. I'd like your eyes on [this](https://github.com/NASA-IMPACT/veda-ui/pull/668/commits/c055024679ba1be231d60cc00a453390a06a6b86) at some point if you have time.
